lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -131,12 +131,13 @@ bool isMarkedWithTpp(linalg::LinalgOp linalgOp, const std::string &target) {
   return libraryCall.compare(target) == 0;
 }
 
-// Returns true if the region of the linalgOp has only a single operation
-// (linalg.yieldOp).
-static bool hasOnlyYieldOp(Region &region) {
+bool hasOnlyYieldOp(Region &region) {
```

Comment:
All of these `hasOnlyOneXXXOp` functions are the same. This should be easily replaceable by a template function on the header.

Reply:
I don't see how I can unify `hasOnlyScalarElementwiseOp` and `hasOnlyYieldOp`. The latter only check for a yield operation while the former for a yield plus an additional operation.

Reply:
Well, if the op is empty, you only check for the yield?

Reply:
I think it will be confusion. `hasOnlyScalarElementwiseOp` requires a region with 2 ops, while `hasOnlyYield` a single op in the region. I am also struggling to find a good name, if we merge the two.

Reply:
No it won't. The logic is simple: yield = one op, others = two ops.

I'd name it `hasOnly` with the type being an indication of what "only" means. Yield is assumed to be present always, so it's not confusing.

Something like:
```
template <class OpTy> bool hasOnly(Region &region);
```

called as:
```
// Here is clear
bool hasOnlyYield = hasOnly<YieldOp>(region);

// Here yield is implied, so also clear
bool hasOnlyAdd = hasOnly<AddOp>(region);
```

Reply:
Got it. But let's do in another commit. It is used in different place.